dcos/istio/README.md

```diff
@@ -0,0 +1,157 @@
+```
+
+## Deploy Kubernetes
+
+This section is based on https://github.com/mesosphere/dcos-kubernetes-quickstart
+
+```bash
+dcos package install --yes beta-kubernetes
+
+# get public IP of DC/OS master
+LB_NAME=$(aws elb describe-load-balancers | jq -r ".LoadBalancerDescriptions[] | select(.DNSName==\"$DCOS_URL\") | .LoadBalancerName")
+INSTANCE_ID=$(aws elb describe-instance-health --load-balancer-name $LB_NAME | jq -r .InstanceStates[0].InstanceId)
+PUBLIC_IP=$(aws ec2 describe-instances --instance-ids $INSTANCE_ID | jq -r .Reservations[0].Instances[0].PublicIpAddress)
+
+# start a tunnel to kubernetes api server
+ssh -N -L 9000:apiserver-insecure.kubernetes.l4lb.thisdcos.directory:9000 core@$PUBLIC_IP
+
+# configure kubectl
+kubectl config set-cluster dcos-k8s --server=http://localhost:9000
+kubectl config set-context dcos-k8s --cluster=dcos-k8s --namespace=default
+kubectl config use-context dcos-k8s
+
+# confirm cluster is operational
+kubectl get nodes
+kubectl --all-namespaces=true get all
+
+# install kube-dns
+kubectl create -f https://raw.githubusercontent.com/mesosphere/dcos-kubernetes-quickstart/master/add-ons/dns/kubedns-cm.yaml
+kubectl create -f https://raw.githubusercontent.com/mesosphere/dcos-kubernetes-quickstart/master/add-ons/dns/kubedns-svc.yaml
+kubectl create -f https://raw.githubusercontent.com/mesosphere/dcos-kubernetes-quickstart/master/add-ons/dns/kubedns-deployment.yaml
+
+# install kubernetes-dashboard
+kubectl create -f https://raw.githubusercontent.com/mesosphere/dcos-kubernetes-quickstart/master/add-ons/dashboard/kubernetes-dashboard.yaml
+open http://localhost:9000/ui # osx only
+```
+
+## Deploy Istio
+
+Note there is a known issue with the beta-kubernentes DC/OS package not
+supporting service accounts. By default the Istio components use service
+accounts to find and connect to the Kubernetes API Server. As a workaround,
+we'll provide modified versions of the Istio and sample app Kubernetes configs
+that include a kubeconfig file, instructing our components how to connect to
+Kubernetes.
+
+### `kubeconfig` file, defined in our istio `ConfigMap` object:
+
+```yaml
+kubeconfig: |-
+  apiVersion: v1
+  kind: Config
+  preferences: {}
+  current-context: dcos-k8s
+
+  clusters:
+  - cluster:
+      server: http://apiserver-insecure.kubernetes.l4lb.thisdcos.directory:9000
+    name: dcos-k8s
+
+  contexts:
+  - context:
+      cluster: dcos-k8s
+      namespace: default
+      user: ""
+    name: dcos-k8s
+```
+
+### Modified Kubernetes config to load a `kubeconfig` file:
+
+```yaml
+spec:
+  containers:
+  - name: proxy
+    image: docker.io/istio/proxy_debug:0.1.6
+    args: ["proxy", "egress", "-v", "2", "--kubeconfig", "/etc/istio/config/kubeconfig"]
+    volumeMounts:
+    - name: "istio"
+      mountPath: "/etc/istio/config"
+      readOnly: true
+  volumes:
+  - name: istio
+    configMap:
+      name: istio
+```
+
+For full details on installing Istio, have a look at
```

Comment:
Since we repeat the istio installation steps below, and we have a custom istio.yaml that we apply, maybe this could say "These instructions are based on the istio installation instructions at https://istio.io/docs/tasks/installing-istio.html."